just_audio/lib/just_audio.dart

```diff
@@ -837,6 +837,16 @@ class AudioPlayer {
       return duration;
     } on PlatformException catch (e) {
       try {
+        // cant connect to servers
+        if (e.code == "-1004" && source is LockCachingAudioSource) {
+          // proxy is offline
+          try {
+            await _proxy._server.close(force: true);
```

Comment:
Wouldn't closing the proxy after this error potentially disrupt other active connections on the proxy? That is unless the -1004 error implies all connections are affected, but it seems possible that the -1004 error could happen on one connection and not another.